src/pip/_internal/index/package_finder.py

```diff
@@ -323,23 +324,15 @@ def filter_unallowed_hashes(
     return filtered
 
 
+@dataclass
 class CandidatePreferences:
```

Comment:
The codebase does assign to these attributes from time to time...

Reply:
_sigh_